src/index.tsx

```diff
@@ -75,24 +77,67 @@ export default class AppIntroSlider<ItemT = any> extends React.Component<
     showPrevButton: false,
     showSkipButton: false,
     bottomButton: false,
+    renderVertically: false,
+    autoPlay: false
   };
   state = {
     width: 0,
     height: 0,
     activeIndex: 0,
   };
   flatList: FlatList<ItemT> | undefined;
+  activeInterval: any;
+
+  componentDidMount() {
+    if (this.props.autoPlay) this.startScroll();
+  }
+
+  // Clear interval when user closes
+  componentWillUnmount() {
+    clearInterval(this.activeInterval);
+  }
+
+  startScroll() {
+    this.activeInterval = setInterval(this.scrolling, 2000);
+  }
+
+  scrolling = () => {
+    if (this.state.activeIndex < this.props.data.length - 1) {
+      this.goToSlide(this.state.activeIndex + 1)
+    } else {
+      clearInterval(this.activeInterval);
+    }
+  }
 
   goToSlide = (pageNum: number, triggerOnSlideChange?: boolean) => {
+    if (this.props.renderVertically) {
+      this.goToSlideV(pageNum, triggerOnSlideChange)
+    }
+    else {
+      this.goToSlideH(pageNum, triggerOnSlideChange)
+    }
+  }
+
+  goToSlideH = (pageNum: number, triggerOnSlideChange?: boolean) => {
     const prevNum = this.state.activeIndex;
-    this.setState({activeIndex: pageNum});
+    this.setState({ activeIndex: pageNum });
     this.flatList?.scrollToOffset({
       offset: this._rtlSafeIndex(pageNum) * this.state.width,
     });
     if (triggerOnSlideChange && this.props.onSlideChange) {
       this.props.onSlideChange(pageNum, prevNum);
     }
   };
+  goToSlideV = (pageNum: number, triggerOnSlideChange?: boolean) => {
```

Comment:
I think it's sufficient with a single `gotToSlide`-function which then uses either `this.state.height` og `this.state.width` depending on the value of `renderVertically`

Reply:
okay i'll change that